### ISSUE TITLE
Spurious CA1305 for Guid.Parse with latest-recommended analysis level in net7.0

### ISSUE THREAD

**Issue by bgrainger** [dotnet/roslyn-analyzers#5999]
### Description

After updating the `TargetFramework` for an existing program that uses `<AnalysisLevel>latest-recommended</AnalysisLevel>` to `net7.0`, analysis starts generating spurious CA1305 warnings for `Guid.Parse`, because it now has an overload that takes `IFormatProvider` (but completely ignores it) https://github.com/dotnet/runtime/blob/af2a48ddaec9a3e27002b0d3b80c07431c2e38c1/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L1490

If `TreatWarningsAsErrors` is `true`, the program no longer compiles.

### Reproduction Steps

Create a new Console app with this csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <AnalysisLevel>latest-recommended</AnalysisLevel>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>

</Project>
```

(This is basically adding `<AnalysisLevel>latest-recommended</AnalysisLevel><TreatWarningsAsErrors>true</TreatWarningsAsErrors>` to the results of `dotnet new console`.

And this source code:

```csharp
Console.WriteLine(Guid.Parse("11223344-5566-7788-99AA-BBCCDDEEFF00"));
```

Run `dotnet run`:

> C:\Code\Projects\GuidParse\Program.cs(1,19): error CA1305: The behavior of 'Guid.Parse(string)' could vary based on the current user's locale settings. Replace this call in '<top-level-statements-entry-point>' with a call to 'Guid.Parse(string, IFormatProvider)'. [C:\Code\Projects\GuidParse\GuidParse.csproj]

If it's changed back to `<TargetFramework>net6.0</TargetFramework>`, the program compiles and runs normally.

### Expected behavior

No CA1305 analysis warning is introduced for existing working code when updating the target framework to `net7.0`, particularly because no benefit is gained by using the suggested overload that takes an `IFormatProvider`.

### Actual behavior

`dotnet run` produces this output and doesn't run the program:

> C:\Code\Projects\GuidParse\Program.cs(1,19): error CA1305: The behavior of 'Guid.Parse(string)' could vary based on the current user's locale settings. Replace this call in '<top-level-statements-entry-point>' with a call to 'Guid.Parse(string, IFormatProvider)'. [C:\Code\Projects\GuidParse\GuidParse.csproj]

### Regression?

Yes; no problem with this code targeting `net6.0`.

### Configuration

7.0.100-preview.4.22252.9
Windows 10, x64

**Comment by bgrainger** [comment]
> I don't think it's desirable / scalable for the analyzer to maintain a blocklist of types that currently ignore the argument.

Don't know if it changes your opinion, but it looks like the analyser _does_ currently maintain a list of types that implement an invariant `ToString`, and counts `Guid` among those types:
https://github.com/dotnet/roslyn-analyzers/blob/d2ef898a6283858c1224c0114cd87fa27b6cb091/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs#L90-L95

**Comment by stephentoub** [comment]
> Don't know if it changes your opinion, but it looks like the analyser does currently maintain a list of types that implement an invariant ToString, and counts Guid among those types:

If it already has one, then yes, it should use it :-)
